### PR TITLE
feat(daemon): be able to use another systemd-module

### PR DESCRIPTION
**Pull request by daylicron**
#### Pull Request (PR) description
- feat(daemon): be able to use another systemd-module
- feat(node_exporter): be able to use another systemd-module

#### This Pull Request (PR) fixes the following issues
n/a

### CODE REVIEW

File: manifests/daemon.pp [hunk 1]
```diff
@@ -79,6 +82,7 @@
   Boolean $purge                       = true,
   String $options                      = '',
   String $init_style                   = $prometheus::init_style,
+  Boolean $deploy_unit_file            = true,
```
Comment (bastelfreak) [inline manifests/daemon.pp:85]: what about making this a generic parameter like `manage_service_file`? Maybe people also want to use their custom sysv file.

File: manifests/node_exporter.pp [hunk 2]
```diff
@@ -167,6 +168,7 @@
     purge              => $purge_config_dir,
     options            => $options,
     init_style         => $init_style,
+    deploy_unit_file   => $deploy_unit_file,
```
Comment (bastelfreak) [inline manifests/node_exporter.pp:171]: this needs to be added to all exporter classes

File: manifests/node_exporter.pp [hunk 3]
```diff
@@ -94,28 +94,29 @@
...
-  String $arch                        = $prometheus::real_arch,
-  String $bin_dir                     = $prometheus::bin_dir,
-  Optional[Array[String]] $collectors = undef,
-  Array[String] $collectors_enable    = [],
-  Array[String] $collectors_disable   = [],
-  Boolean $export_scrape_job          = false,
-  Stdlib::Port $scrape_port           = 9100,
-  String[1] $scrape_job_name          = 'node',
-  Optional[String[1]] $bin_name       = undef,
+  Boolean $purge_config_dir             = true,
```
Comment (bastelfreak) [inline manifests/node_exporter.pp:95]: the changed indentation isn't required in this file. please undo it.

**Comment by bastelfreak** [comment]
Hi @daylicron. can you please document this in the README.md and add tests?

File: manifests/node_exporter.pp [hunk 4]
```diff
@@ -86,36 +86,37 @@
 #  The binary release version
 
 class prometheus::node_exporter (
-  String $download_extension,
```
Comment (bastelfreak) [inline manifests/node_exporter.pp:89]: Those two parameters are mandatory. They are also defined in `data/defaults.yaml`.

**Comment by alexjfisher** [comment]
Is https://github.com/voxpupuli/puppet-extlib/blob/master/REFERENCE.md#extlibhas_module a possibility here?

**Comment by anarcat** [comment]
would love to see this happen. it's one of the things I am trying to do (poorly) in #303. 

there I called the parameter `manage_systemd_unit`, but maybe `manage_service_file` would make more sense. it's a harder patch to implement though because it affects *all* init systems instead of just systemd.

**Comment by daylicron** [comment]
I'm sorry, I obviously don't find any time to put the necessary work in this PR. By now we already switched our Puppet SystemD module to camptocamp-systemd. Therefore I'm closing this PR. If anybody has the need to implement the feature, feel free to use this as a base.

**Pull request closed without merging**